docs/examples/DataSet/DataSet-class-walkthrough.ipynb

```diff
@@ -0,0 +1,1327 @@
+    }
+   ],
+   "source": [
+    "plot_dataset(dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Note on immutability of DataSet\n",
+    "\n",
+    "Users are NOT expected to use `DataSet`s methods that modify its content, for example, `add_result`. Only `Measurement` and `DataSaver` objects use those methods. This is due to the phylosophy of the `DataSet` - once created within a `Measurement` context, it should not change so that the user don't accidentally modify/remove the important measured data. In the future versions of QCoDeS the design of the `DataSet` object will be improved to make the described phylosophy clear."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## DataSet indentity\n",
+    "\n",
+    "Before we dive into what's in the `DataSet`, let's briefly note how a `DataSet` is identified.\n",
+    "\n",
+    "TL;DR:\n",
+    "> * Both `experiment name` and `sample name` are great means for identifying a dataset, especially when coupled with the `captured_run_id`.\n",
+    "> * `guid` is the only true globally unique identifier for datasets."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### GUID\n",
+    "\n",
+    "Globally, \"in this universe\", a dataset is identified by a globally unique identifier:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'aaaaaaaa-0000-0000-0000-016e037351ff'"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset.guid"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`guid` is generated when a dataset is created, and is globally unique.\n",
+    "\n",
+    "Due to the nature of the `guid` and its size, it is very user-friendly. For example, it is convenient use it as a reference in a conversation betweenusers. However, the `guid` is crutial for use in automated/computer systems, for example, when moving a dataset from one database file to a different one."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Captured run ID (and run ID)\n",
+    "\n",
+    "Within a given single database file (!) a dataset can be identified by its `run_id`, an integer index:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "3"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset.run_id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`run_id` gets assigned to a dataset upon its creation, and is unique only within that database file.\n",
```

Comment:
I think it would be better to do the full explanation via captured_run_id and then either not mention run_id or only mention it in a note at the end.

Reply:
Same goes for counter

Reply:
ok, will do